core/src/toga/widgets/slider.py

```diff
@@ -194,48 +221,156 @@ def on_change(self, handler):
         self._on_change = wrapped_handler(self, handler)
 
     @property
-    def on_press(self):
-        """The function for when the user click the slider before sliding it.
-
-        Returns:
-            The ``callable`` that is executed when the slider is clicked.
-        """
+    def on_press(self) -> callable:
+        """Handler to invoke when when the user presses the slider before changing it."""
         return self._on_press
 
     @on_press.setter
     def on_press(self, handler):
         self._on_press = wrapped_handler(self, handler)
 
     @property
-    def on_release(self):
-        """The function for when the user release the slider after sliding it.
-
-        Returns:
-            The ``callable`` that is executed when the slider is released.
-        """
+    def on_release(self) -> callable:
+        """Handler to invoke when the user releases the slider after changing it."""
         return self._on_release
 
     @on_release.setter
     def on_release(self, handler):
         self._on_release = wrapped_handler(self, handler)
 
-    @property
-    def on_slide(self):
-        """The function for when the value of the slider is changed.
 
-        **DEPRECATED: renamed as on_change**
+class SliderImpl(ABC):
+    @abstractmethod
+    def get_value(self):
+        ...
 
-        Returns:
-            The ``callable`` that is executed on slide.
-        """
-        warnings.warn(
-            "Slider.on_slide has been renamed Slider.on_change", DeprecationWarning
-        )
-        return self._on_change
+    @abstractmethod
+    def set_value(self, value):
+        ...
+
+    @abstractmethod
+    def get_range(self):
+        ...
+
+    @abstractmethod
+    def set_range(self, range):
+        ...
+
+    @abstractmethod
+    def get_tick_count(self):
+        ...
+
+    @abstractmethod
+    def set_tick_count(self, tick_count):
+        ...
+
+
+class IntSliderImpl(SliderImpl):
+    """Base class for implementations which use integer values."""
+
+    # Number of steps to use to approximate a continuous slider.
+    CONTINUOUS_MAX = 10000
+
+    def __init__(self):
+        super().__init__()
+
+        # Dummy values used during initialization.
+        self.value = 0
+        self.discrete = False
+
+    def get_value(self):
+        return self.value
```

Comment:
Is there any reason to prefer a shadow copy here rather than using `self.interface._round_value(self.get_int_value())`?

Reply:
It ensures that the value is round-trippable on all platforms, which could be important for some apps.